config/routes.rb

```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  root "users#index"
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get '/users/:id', to: 'users#show', as: :user
+
+  get '/users/:user_id/posts', to: 'posts#index', as: :posts
+
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: :post
```

Comment:
- [ ] **[OPTIONAL]** You can define these route more efficiently using the `resources` helper as provided in Rails.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```